File: CompilationTest/ConstrInferBug/results/ConstrInferBug.cs
public class TestClass
{
    public void MethodTest(List<string> methodParam)
    {
    }
    
    public void TestMethod()
    {
    }
}

public class Program
{
    static public void Main(string[] args)
    {
        try 
        {
            new TestClass().TestMethod();
        }
        catch (System.Exception e)
        {
            System.Console.WriteLine($"Exception: {e.Message}");
        }
    }
}

File: CompilationTest/ConstrInferBug/results/ConstrInferBug.cpp
#include <OneLang-Core-v0.1/one.hpp>

class TestClass {
  public:
    void methodTest(one::vec<std::string> method_param) {
    }
    
    void testMethod() {
    }

  private:
};

int main()
{
    try {
        TestClass c;
        c.testMethod();
    } catch(std::exception& err) {
        std::cout << "Exception: " << err.what() << '\n';
    }
    return 0;
}